src/main/keystone.ts

```diff
@@ -0,0 +1,90 @@
+import { dungeonTimersByMapId } from "./constants";
+
+enum VideoSegmentType {
+    BossEncounter = 'Boss',
+    Trash = 'Trash'
+};
+
+class ChallengeModeVideoSegment {
+    logEnd: Date
+    result?: Boolean
+
+    constructor(
```

Comment:
Yeah, it's a short hand constructor that takes arguments and makes them as public members of the class as well. It's the same result as the following, in this particular case, but much more concise:

```typescript
class ChallengeModeVideoSegment {
    logEnd: Date;
    result?: Boolean;
    segmentType: VideoSegmentType;
    logStart: Date;
    ts: number;
    encounterId?: number;

    constructor(
        segmentType: VideoSegmentType,
        logStart: Date,
        ts: number,
        encounterId?: number
    ) {
        this.segmentType = segmentType;
        this.logStart = logStart;
        this.ts = ts;
        this.encounterId = encounterId;

        // Initially, let's set this to log start date to avoid logEnd
        // potentially being undefined.
        this.logEnd = logStart;
    }
};
```

Reply:
neat - thanks for the education!